clean-abap/CleanABAP_es.md

```diff
@@ -0,0 +1,5127 @@
+
+Si tienes suficiente espacio, escribe los nombres completos.
+Comienza a abreviar únicamente si excedes el límite de caracteres.
+
+Si tienes que abreviar, comienza con las palabras con _poca importancia_.
+
+Abreviar puede parece eficiente inicialmente, pero se vuelve ambiguo muy rápido.
+Por ejemplo, nombrar una variable como `cust` significa "customizing", "customer" o "custom"?
+Las tres son muy comunes en aplicaciones SAP.
+
+> Lee más en _Capítulo 2: Nombres con sentido: Realizar distinciones con sentido_ de [_Código Limpio_ por Robert C. Martin].
+
+### Usa las mismas abreviaciones en todas partes
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Nomenclatura](#nomenclatura) > [Esta sección](#usa-las-mismas-abreviaciones-en-todas-partes)
+
+Las personas buscan usando palabras claves para encontrar código relevante.
+Apoya esto usando la misma abreviación para el mismo concepto.
+Por ejemplo, siempre abrevia "tipo de objeto de detección" a "tipobjdet",
+en lugar de mezclar "tod", "tipod", "tipobjd", etc.
+
+> Lee más en _Capítulo 2: Nombres con sentido: Usar nombres que se puedan buscar_ de [_Código Limpio_ por Robert C. Martin].
+
+### Usa sustantivos para las clases y verbos para los métodos
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Nomenclatura](#nomenclatura) > [Esta sección](#usa-sustantivos-para-las-clases-y-verbos-para-los-métodos)
+
+Usa sustantivos o frases con sustantivos para nombrar clases, interfaces y objetos:
+
+```ABAP
+CLASS /clean/account
+CLASS /clean/user_preferences
+INTERFACE /clean/customizing_reader
+```
+
+Usa verbos o frases de verbos para nombrar métodos:
+
+```ABAP
+METHODS withdraw
+METHODS add_message
+METHODS read_entries
+```
+Iniciar métodos booleanos con verbos como `is_` y `has` provee un flujo de lectura agradable:
+
+```ABAP
+IF is_empty( table ).
+```
+Recomendamos nombrar las funciones como a los métodos:
+
+```ABAP
+FUNCTION /clean/read_alerts
+```
+
+### Evita palabras poco específicas como "data", "info", "object"
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Nomenclatura](#nomenclatura) > [Esta sección](#evita-palabras-poco-específicas-como-data-info-object)
+
+Omite palabras que generan ruido:
+
+```ABAP
+account  " instead of account_data
+alert    " instead of alert_object
+```
+o reemplazalas con algo específico que realmente agregue valor
+
+```ABAP
+user_preferences          " instead of user_info
+response_time_in_seconds  " instead of response_time_variable
+```
+
+> Lee más en _Capítulo 2: Nombres con sentido: Realizar distinciones con sentido_ de [_Código Limpio_ por Robert C. Martin]
+
+### Elige una palabra por concepto
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Nomenclatura](#nomenclatura) > [Esta sección](#elige-una-palabra-por-concepto)
+
+```ABAP
+METHODS read_this.
+METHODS read_that.
+METHODS read_those.
+```
+
+Elige un término para un concepto y apégate a él; no lo mezcles usando otros sinónimos.
```

Comment:
```suggestion
Elige un término para un concepto y cíñete a él; no lo mezcles usando otros sinónimos.
```

Reply:
I've never heard the word "ceñir" being used as a replacement for "apego", I guess this is also due to differences in Spanish. I'd suggest using some other word that conveys the same meaning, but might be more universal like "adherir".

```suggestion
Elige un término para un concepto y adhiérete a él; no lo mezcles usando otros sinónimos.
```

Reply:
Ceñirse means stick to something and not deviate (e.g. ceñirse al plan). Adherirse would also be good, but apegarse definitely not because it means emotional attachment.

Reply:
You're right, but it's very common at least in Mexico. I guess it's one of those cases where language is misused so often it becomes the norm. I'll leave "adherirse" as I'm sure "ceñirse" would confuse readers that are not from Spain.

Reply:
Agreed, it is probably a good middle ground.